consensus/src/main/java/org/ethereum/beacon/consensus/spec/ForkChoice.java

```diff
@@ -109,7 +110,7 @@ default Hash32 get_head(Store store) {
       }
 
       head = children.stream()
-          .max(Comparator.comparing(root -> get_latest_attesting_balance(store, root)))
+          .max(Comparator.comparing(root -> Pair.with(get_latest_attesting_balance(store, root), root)))
```

Comment:
Let's make sure that implementation of `BytesValue.compareTo` is aligned with the spec.

Reply:
Compared to python.
Python spec implementation uses bytes (python type) to hold hash32 values.
They are compared with a lexicographical order which matches BytesValue implementation.
E.g. from left to right, unsigned bytes.
Added a test for that.